lib/ecto/uuid.ex

```diff
@@ -35,12 +35,12 @@ defmodule Ecto.UUID do
   def cast(_), do: :error
 
   @doc """
-  Same as `cast/1` but raises `Ecto.CastError` on invalid arguments.
+  Same as `cast/1` but raises `Ecto.Type.CastError` on invalid arguments.
   """
   def cast!(value) do
```

Comment:
while at it, I found myself re-implementing `cast!` on pretty much in all my custom ecto types, repeating the pattern:

```elixir
def cast!(value)
  case cast(value) do
    {:ok, value} -> value
    :error -> raise "cannot cast ... as ..."
  end
```

often. WDYT about introducing `Ecto.Type.cast!`? Then, we could have:

```elixir
defmodule Ecto.UUID do
  # ...

  def cast!(value), do: Ecto.Type.cast!(__MODULE__, value)
end
```

if sounds like worth pursuing, I can open up thread/issue/pr to flesh it out (there's a caveat that: `Ecto.Type.cast(Ecto.UUID, nil) => {:ok, nil}`, but `Ecto.UUID.cast(nil) => :error` so not sure if it can be easily made generic in Ecto.Type)